pkgs/emacs/wrapper.nix

```diff
@@ -69,6 +71,36 @@
     );
     executablePackages = map (pkg: "${pkg}/bin") executablePackages;
   });
+
+  macosAppLauncher = writeShellScript "emacs-launcher" ''
+    SCRIPT_DIR="$( cd "$( dirname "''${BASH_SOURCE[0]}" )" && pwd )"
+    APP_CONTENTS="$(dirname "$SCRIPT_DIR")"
+    APP_RESOURCES="$APP_CONTENTS/Resources"
+
+    # Use a login shell to source the Nix environment
+    # This ensures we get the same environment
+    exec -l "$SHELL" -c '
+      if [ -d "'$APP_RESOURCES'/site-lisp" ]; then
+        export EMACSLOADPATH="'$APP_RESOURCES'/site-lisp:"
+      fi
+
+      if [ -d "'$APP_RESOURCES'/info" ]; then
+        export INFOPATH="'$APP_RESOURCES'/info:${emacs}/share/info''${INFOPATH:+:$INFOPATH}"
```

Comment:
Do you need single quotes inside double quotes? Looks weird

Reply:
Yeah, we need the single quotes (`'$APP_RESOURCES'`) because we have the outer single quotes we pass in `exec -l "$SHELL" -c ''` - these single quotes around `'$APP_RESOURCES'`, etc., ensure that those variables are expanded within the outer single quotes. So, they are needed